### ISSUE TITLE
Debian is not Linux

### ISSUE THREAD

**Issue by ShalokShalom** [webtorrent/webtorrent-desktop#1584]
Currently, the Linux download link links to the deb file. Why not providing the tar.xz here? You could also add the .deb as Debian/Ubuntu Link.

**Comment by pro-src** [comment]
This reads as "Car is not Vehicle".

Debian is a distribution of the Linux operating system. Why not call it Linux? It is a Linux. As you'd expect when looking for Linux support, you found a `.deb`. I completely agree with adding packages for other *nix. Until that happens, the current link behavior is reasonable. It reflects the popularity of Debian based distros.

**Comment by pro-src** [comment]
***Here is how to produce your own:***
* Install [Node.js](https://nodejs.org/en/download/)
* Install NPM if missing from Node.js installation: https://github.com/npm/cli#fancy-install-unix 
* `git clone https://github.com/webtorrent/webtorrent-desktop`
* `cd webtorrent-desktop`
* `npm install`
* `npm run package -- linux --package=zip`
* `cd dist`

***The zip files will be in the current (dist/) directory, proceed if you want a compressed tarball:***
* `tar -zcvf WebTorrent-x64.tar.gz WebTorrent-linux-x64`
* `tar -zcvf WebTorrent-ia32.tar.gz WebTorrent-linux-ia32`

***Test run from (dist/) directory:***
* `cd WebTorrent-x64`
* `./WebTorrent`

**Comment by ShalokShalom** [comment]
Thanks a lot. May you add that to the Readme?

Treating Debian as the only Linux distribution results in the experience, that one clicks on the Linux button and getting something that does not work for him/her.

It is like saying that all cars are roadsters or that there are only trucks and people who are looking for a combi, get something completely not working for them.

You could also argue that both Debian and Windows are an operating system, so simply describe the button as "Installer and give them an exe.

**Comment by pro-src** [comment]
Yw, I was thinking that the `package.js` could be updated to add `tar.gz` as a package type in the same PR. I might get around to fiddling with it but I'm not making any reservations. If you want to send one, I think that would be great.

> Treating Debian as the only Linux distribution results in the experience, that one clicks on the Linux button and getting something that does not work for him/her.

I agree but it likely doesn't apply to the majority of users.

> You could also argue that both Debian and Windows are an operating system, so simply describe the button as "Installer and give them an exe.

I don't think that's a good comparison because the current behavior actually makes some sense...
I'm only picking at the issue title and providing explanation for the current behavior.

I would reopen this issue until it's been addressed by maintainers.

**Comment by ShalokShalom** [comment]
>  I agree but it likely doesn't apply to the majority of users.

This could also be said about Linux in general. Just ship Windows executables, since not shipping Linux ones does not affect the majority of users. Guess why you dont have much non-.deb based users in the first place. 

> I don't think that's a good comparison because the current behaviour actually makes > some sense... 

How does it make any sense on any non-Debian based Distro?

**Comment by pro-src** [comment]
<details><summary>:thinking: (spoiler)</summary>

I prefer https://get.adobe.com/flashplayer/ over https://get.adobe.com/reader/otherversions/.
Wouldn't it make the most sense for a `.deb` to be the selected default on https://get.adobe.com/flashplayer/ for Linux?
If I'm not making assumptions about the technical experience of users, I would.

This is arguably a feature when you consider that Linux newbies:
* Are most likely to be running a Debian based distro.
* Aren't likely to be sure of which package they need or know about the compressed tarball format.
* Seasoned users of Linux are much less likely to be deterred from seeking out installation instructions.

IMHO, The current behavior is the best behavior for the majority of users regardless of whether they are new to Linux or not. In the beginning, this was likely the best behavior for marketing the project.

That is my unbiased opinion. I don't actually like the current behavior.
</details>

**Comment by pro-src** [comment]
Being fair, there is an "Other Downloads" link.
![webtorrent io_desktop_](https://user-images.githubusercontent.com/34285059/58758918-49fb5d00-84e8-11e9-9e1f-3cfc0f854e45.png)

**Comment by ShalokShalom** [comment]
Preferring a distro like that is just a smack in the face of any other distribution and considered as a shady practice in the Linux and open source communities. People are allowed to run whatever they want, and by making deb the default by an *independent* distribution, you are just making it that much harder for people to try other distributions.

**Comment by pro-src** [comment]
I don't see it that way. If not opinion, where are the stats to backup your claims?

> This could also be said about Linux in general. Just ship Windows executables, since not shipping Linux ones does not affect the majority of users. Guess why you dont have much non-.deb based users in the first place.

For example, how do you know that the majority of WebTorrent desktop users are running Windows?

**Comment by ShalokShalom** [comment]
It's not the case. The thing is, that the assumption that one OS/Distribution is worth treating as a first class citizen just because it has the most users, is all just not free software.

**Comment by pro-src** [comment]
Thanks for the interesting discussion. It was very pleasant. Hopefully, they'll get around to addressing this issue.

**Comment by Borewit** [comment]
I concur it would be better to describe the Debian package as a Debian package.

It would be great to offer additional installers like:
* RPM support (#1502)
* AppImage (#1102)

If you want the change, get ready participate gents.

**Comment by ShalokShalom** [comment]
Hi :hugs: 

So what exactly stops us from shipping an AppImage since 2 years?

You could also implement an auto-updater for Linux, this way. 

It works perfectly here. 

Just make the AppImage which is provided in the linked issue executable and click it.

Easy peasy.

**Comment by Borewit** [comment]
> So what exactly stops us from shipping an AppImage since 2 years?

Same story my friend.

Someone taking the lead and push for it by creating a PR, a clear description, review it yourself, make screenshots make sure others can review it easy.

**Comment by ShalokShalom** [comment]
:eyes:

**Comment by feross** [comment]
I don't think that users of non-Debian Linux will have difficulty figuring out how to install from the .zip files provided at the "Other Downloads" link. The link is there mainly to benefit Ubuntu users who (of all the Linux users out there) are the most likely to be unable to figure out which install file to use on their own.

I am in favor of offering more better linux install options, but I don't think we should add more than one download link for Linux. As Debian/Ubuntu are the most popular distribution, they get the link.

**Comment by feross** [comment]
I think the right solution is probably to make the "Other Downloads" link point to a page on the WebTorrent site which lists out install instructions for each platform, including all of the Linux flavors. And we can keep the big simple Linux button for Ubuntu (and perhaps include a small note that it's actually for Ubuntu/Debian)

**Comment by Borewit** [comment]
Why not a download for Linux button which access a second page with specialized instructions / buttons for each flavour?

Maybe an autodetect which directly jumps to the page offset corresponding to the OS browsing the download page.

Click Linux, click flavour, done.

**Comment by feross** [comment]
That's also quite reasonable.

**Comment by chrisdebian** [comment]
Does flathub.org help with this?

Thanks,

Chris
chris_debian
